### PR TITLE
added HTML-CSS widget 

### LINKED ISSUE #2861: Feature request - HTML viewer widget
Feature Request : 
Current scenario - There is no widget, which will show an HTML page/ window upon providing raw HTML data

What we can do to add the above functionality - 
We can add a feature where users can provide raw HTML and can see the output in the application itself using the HTML viewer widget

### PR DESCRIPTION
**Pull request by manishkushare**
Resolve #2861

### CODE REVIEW

File: frontend/src/Editor/Components/Html.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  const css = stringifyCSS
+    .split('}')
+    .reduce((acc, cv) => {
+      if (cv) {
+        acc.push(`.custom_css ${cv}}`);
+      }
+      return acc;
+    }, [])
+    .join('');
+  console.log(css);
```
Comment (gsmithun4) [inline frontend/src/Editor/Components/Html.jsx:15]: remove console log

File: frontend/src/Editor/Components/Html.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  const baseStyle = stringifyCSS
+    ? null
+    : {
+        backgroundColor: darkMode ? '#47505D' : '#ffffff',
+        color: darkMode ? 'white' : 'black',
+      };
+  console.log(baseStyle);
```
Comment (gsmithun4) [inline frontend/src/Editor/Components/Html.jsx:22]: remove console log

File: frontend/src/Editor/Components/Html.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  const { visibility } = styles;
+
+  const [rawHtml, setRawHtml] = useState('');
+  useEffect(() => {
+    console.log(css);
```
Comment (gsmithun4) [inline frontend/src/Editor/Components/Html.jsx:27]: remove console log

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Comment by sherfin94** [comment]
Hey @manishkushare , it seems like the raw css feature is not working.
image

**Comment by gsmithun4** [comment]
> Hey @manishkushare , it seems like the raw css feature is not working. image

@sherfin94 @manishkushare Is it possible to use code hinter type input for HTML and CSS?

**Comment by manishkushare** [comment]
[image: Screenshot 2022-05-03 at 3.28.58 PM.png]

**Comment by manishkushare** [comment]
@sherfin, we don't need to put string for color:''red', we can simply write
color:red

**Comment by manishkushare** [comment]
@gsmithun4 @sherfin94, we don't need to provide color:"red" instead, we can
simply write color:red. Since we are expecting it as raw CSS and also
according to CSS rule, we generally write color : red, for React we need to
write color:"red"

Also, yes we can think about the code hinter input type, Last time, I had a
conversation with Arpit, and we thought that, since we expecting raw HTML
and CSS from the user, we can keep the input field as it is, and we also
have code hinter widget, where user can write code into it and pass that
value as raw HTML inside HTML widget

**Comment by manishkushare** [comment]
@gsmithun4 I can try adding code hinter as an input field.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
Hey @manishkushare , could you resolve conflicts and merge the latest develop into this and add documentation?

**Review by arpitnath — CHANGES_REQUESTED** [review]
@manishkushare Please fix the eslint errors and warnings

**Comment by arpitnath** [comment]
> @manishkushare Please fix the eslint errors and warnings


@sherfin94 Ignore my comment, will fix the eslint issue in a separate PR.

**Comment by arpitnath** [comment]
> @manishkushare Please fix the eslint errors and warnings


@sherfin94 Ignore my comment, will fix the eslint issue in a separate PR.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
@manishkushare As we discussed, lets remove the style property for now.

Also, it would be nice to fix the lint as @arpitnath suggested. Only some minor linting issues are there.

**Review by saingsocheat — APPROVED** [review]

**Pull request merged**